app_dart/cron.yaml

```diff
@@ -6,30 +6,43 @@ cron:
 - description: refresh chromebot build status
   url: /api/refresh-chromebot-status
   schedule: every 3 minutes
+
 - description: retrieve missing commits
   url: /api/vacuum-github-commits
   schedule: every 1 hours
-- description: sends build status to GitHub to annotate PRs and commits
+
+- description: sends build status to GitHub to annotate flutter PRs and commits
   url: /api/push-build-status-to-github?repo=flutter/flutter
   schedule: every 1 minutes
-- description: sends pr-specific gold status to GitHub to annotate PRs and commits
-  url: /api/push-gold-status-to-github
-  schedule: every 5 minutes
+
 - description: sends build status to GitHub to annotate engine PRs and commits
   url: /api/push-build-status-to-github?repo=flutter/engine
   schedule: every 2 minutes
+
+- description: sends pr-specific gold status to GitHub to annotate flutter PRs and commits
+  url: /api/push-gold-status-to-github?repo=flutter/flutter
+  schedule: every 5 minutes
+
+- description: sends pr-specific gold status to GitHub to annotate engine PRs and commits
+  url: /api/push-gold-status-to-github?repo=flutter/engine
+  schedule: every 5 minutes
```

Comment:
Having separate cron jobs are probably less ideal.
Gold already has a very tight budget for the requests we make to it, so if these were running simultaneously, we might go over.
We should do this more like `check_for_waiting_pull_requests.dart`, with one cron job running through the supported repos.

Here is where that is happening:
https://github.com/flutter/cocoon/blob/main/app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart#L44

Reply:
We send a separate request to Gold for each PR, right? If that's the case, then the number of cron jobs doesn't matter. If there are 10 PRs in flutter and 10 PRs in the engine, we'll be sending 20 requests to Gold regardless of whether we have a single cron job or separate ones. I might be missing something here, so please correct me if I'm wrong :)

Reply:
> We send a separate request to Gold for each PR, right?

We do, but one at a time.

There is a strict time threshold on those Gold endpoints. So the quantity does not matter 10 framework + 10 engine PRs = 20, but if they are running in parallel like this, they could max out that rate limit.

Reply:
Looking at the code, I see the requests being made sequentially anyway (unless I'm misunderstanding that code): https://github.com/flutter/cocoon/blob/c9a5c6d022f2067cd9d8bce52372e729d2901cbb/app_dart/lib/src/request_handlers/push_gold_status_to_github.dart#L53-L55

I'm happy to unify the cron jobs into one. Just trying to fully understand the reason behind it :)

Reply:
Yes, they are done sequentially. But if you have two cron jobs, you will have two running simultaneously.

It is the requests we make to the Gold service itself (not github or the datastore) that are so restricted. 
https://github.com/flutter/cocoon/blob/c9a5c6d022f2067cd9d8bce52372e729d2901cbb/app_dart/lib/src/request_handlers/push_gold_status_to_github.dart#L203

Since it is a separate Gold instance, it may not be a problem, but it causes a lot of issues if we hit that rate limit.
I never noticed the build status were separate cron jobs, so if it fits the pattern and does not put us over the limit, I am fine with keeping them separate. :)

Reply:
Done.

Makes perfect sense now, thanks for explaining!